plugins/inputs/http_response/http_response.go

```diff
@@ -148,6 +154,7 @@ func (h *HTTPResponse) httpGather() (map[string]interface{}, error) {
 
 	fields["response_time"] = time.Since(start).Seconds()
 	fields["http_response_code"] = resp.StatusCode
+	fields["connection_failed"] = 0
```

Comment:
My thoughts:
It looks like we have 3 different fields that behave similarly, connection_failed, response_timeout, and response_string_match. This feels wrong to me. All three fields are exclusive of each other. You'll never have more than one of them true. So it seems like we should instead have one field which indicates the result. Something like `result_type` with the values `connection failed`/`timeout`/`response match`.

If we do keep them separate fields, I would at least argue that they should be booleans, not ints.